src/cranbtree.c

```diff
@@ -245,7 +261,14 @@ int cbt_key_search(cranbtree_t * cbt, void *object)
 void *cbt_search(cranbtree_t * bt, int key)
 {
 	assert(bt != NULL);
-	return bt_search_helper(bt->root, key, bt->n);
+	void *object = bt_search_helper(bt->root, key, bt->n);
+	
+	if(cbt_key_search(bt, object)== (-1)){
```

Comment:
you should check if object is NULL, and if it is than the object was not found. Dont call cbt_key_search here since it's not yet implemented. i.e if(object == NULL). Leave the return as it is. just set the error code if the condition is satisified